### PR TITLE
Minor architecture docs tweaks

### PR DESCRIPTION
**Pull request by awly**
- consistently use "certificate" instead of "public key"
- make diagram in "local users" section match the text (user "sandra"
  doesn't have access to "grav-02")
- de-duplicate docs on session streaming between auth and proxy pages

### CODE REVIEW

File: docs/4.2/architecture/teleport_architecture_overview.md [hunk 1]
```diff
@@ -219,11 +218,11 @@ $ tsh --proxy=p scp -P example.txt user@host/destination/dir
 ```
 
 Unlike `ssh`, `tsh` is very opinionated about authentication: it always uses
-auto-expiring keys and it always connects to Teleport nodes via a proxy.
+auto-expiring certificates and it always connects to Teleport nodes via a proxy.
 
-When `tsh` logs in, the auto-expiring key is stored in `~/.tsh` and is valid for
-12 hours by default, unless you specify another interval via the `--ttl` flag
-(capped by the server-side configuration).
+When `tsh` logs in, the auto-expiring certificate is stored in `~/.tsh` and is
```
Comment (benarent) [inline docs/4.2/architecture/teleport_architecture_overview.md:223]: It's good to have some fresh eyes on these docs. +1 for using certificates vs keys

File: docs/4.2/admin-guide.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ $ sudo mkdir -p /var/lib/teleport
 
 ### Teleport Checksum
 
-Gravitational Teleport provides a checksum from the 
+Gravitational Teleport provides a checksum from the
```
Comment (benarent) [inline docs/4.2/admin-guide.md:37]: It's a bit of pain / makes it harder to review when removing trailing spaces, but I think it's a good practice and something we should write as part of our docs checks.

File: docs/4.2/architecture/teleport_auth.md [hunk 3]
```diff
@@ -218,41 +220,6 @@ storage.
...
-   required because without it, a proxy will not be able to establish the 2nd
-   connection to the destination node.
-
-However, there are advantages of proxy-based session recording too. When
-sessions are recorded at the nodes, a root user can add iptables rules to
-prevent sessions logs from reaching the Auth Server. With sessions recorded at
-the proxy, users with root privileges on nodes have no way of disabling the
-audit.
-
-See the [admin guide](../admin-guide.md#recorded-sessions) to learn how to turn on the
```
Comment (benarent) [inline docs/4.2/architecture/teleport_auth.md:253]: Is this link still there in teleport_proxy.md ? We should also make clear that the setting is defined on the auth server vs proxy.

**Review by benarent — CHANGES_REQUESTED** [review]
Left a few comments, one change would be to make sure that we still link to the admin guide for how to setup recording mode.

File: docs/4.2/architecture/teleport_auth.md (hunk 3 above)
Comment (awly) [reply docs/4.2/architecture/teleport_auth.md:253]: Yep, the link is there (the section is bit-to-bit identical).
Added a note in `teleport_proxy.md` that proxy recording is configured on auth server

File: docs/4.2/admin-guide.md (hunk 2 above)
Comment (awly) [reply docs/4.2/admin-guide.md]: Ooops, didn't realize how many diffs that produced.
Reverted.

**Review by awly — COMMENTED** [review]
PTAL

**Comment by awly** [comment]
teleport_auth.md still links to https://gravitational.com/teleport/docs/admin-guide/#audit-log from "Audit Log" section, which has docs on configuring recording mode.
Do you think that's sufficient or should I add another, more direct link?

**Review by benarent — APPROVED** [review]
Looks good 👍

**Comment by awly** [comment]
@gravitational-jenkins retest please

**Comment by awly** [comment]
@gravitational-jenkins retest this please

**Comment by awly** [comment]
@gravitational-jenkins retest this please

**Comment by awly** [comment]
@gravitational-jenkins retest this please

**Comment by awly** [comment]
@gravitational-jenkins retest this please

**Pull request merged**